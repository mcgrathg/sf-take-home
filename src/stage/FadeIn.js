import React from 'react';
import PropTypes from 'prop-types';
import { Spring } from 'react-spring/renderprops.cjs';

const FadeIn = ({ children }) => {
  return (
    <Spring
      from={{ opacity: 0 }}
      to={{ opacity: 1 }}
      config={{ duration: 500 }}
    >
      {props => <div style={props}>{children}</div>}
    </Spring>
  );
};

FadeIn.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.arrayOf(PropTypes.object),
  ]).isRequired,
};

export default FadeIn;
