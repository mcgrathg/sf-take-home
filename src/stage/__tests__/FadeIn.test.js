import React from 'react';
import { render } from 'react-testing-library';
import FadeIn from '../FadeIn';

test('renders', () => {
  const { container } = render(
    <FadeIn>
      <div>something</div>
    </FadeIn>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
