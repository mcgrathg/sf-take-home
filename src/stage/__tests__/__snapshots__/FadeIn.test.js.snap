// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  style="opacity: 0;"
>
  <div>
    something
  </div>
</div>
`;
